src/main/js/controls/Line/helpers/helpers.js

```diff
@@ -346,11 +346,11 @@ const processDataPoints = (graphConfig, dataTarget) => {
     dataTarget.style = {
         strokeDashArray: getStrokeDashArray(dataTarget.style)
     };
-    dataTarget.legendOptions = getDefaultLegendOptions(graphConfig, dataTarget);
     dataTarget.showShapes = getDefaultValue(
         dataTarget.showShapes,
         graphConfig.showShapes
     );
+    dataTarget.legendOptions = getDefaultLegendOptions(graphConfig, dataTarget);
```

Comment:
I don't think order is important in here. In this method we are just initiating all the properties given by consumer. Since logic in getDefaultLegendOptions method depends on showShapes, i have to move showShapes before legendOptions.

Reply:
Can you add a comment on the code
```js
\\ Since logic in getDefaultLegendOptions method depends on showShapes, so `getDefaultLegendOptions` should be called after  showShapes
```